bin/db_patcher.js

```diff
@@ -3,211 +3,29 @@
-var database = options.master.database
-delete options.master.database
-
-var client = mysql.createConnection(options.master)
-
-createDatabase()
-  .then(changeUser)
-  .then(checkDbMetadataExists)
-  .then(readDbPatchLevel)
-  .then(patchToRequiredLevel)
-  .then(closeAndReconnect)
-  .done(
-    function() {
-      log.info('Patching complete')
-    },
-    function(err) {
-      log.fatal(err)
-      process.exit(2)
-    }
-  )
-
-// helper functions
-function createDatabase() {
-  var d = P.defer()
-  log.trace( { op: 'MySql.createSchema:CreateDatabase' } )
-  client.query(
-    'CREATE DATABASE IF NOT EXISTS ' + database + ' CHARACTER SET utf8 COLLATE utf8_unicode_ci',
-    function (err) {
-      if (err) {
-        log.error({ op: 'MySql.createSchema:CreateDatabase', err: err.message })
-        return d.reject(err)
-      }
-      d.resolve()
-    }
-  )
-  return d.promise
-}
-
-function changeUser() {
-  var d = P.defer()
-  log.trace( { op: 'MySql.createSchema:ChangeUser' } )
-  client.changeUser(
-    {
-      user     : options.master.user,
-      password : options.master.password,
-      database : database
-    },
-    function (err) {
-      if (err) {
-        log.error({ op: 'MySql.createSchema:ChangeUser', err: err.message })
-        return d.reject(err)
-      }
-      d.resolve()
-    }
-  )
-  return d.promise
-}
-
-function checkDbMetadataExists() {
-  log.trace( { op: 'MySql.createSchema:CheckDbMetadataExists' } )
-  var d = P.defer()
-  var query = "SELECT COUNT(*) AS count FROM information_schema.TABLES WHERE table_schema = ? AND table_name = 'dbMetadata'"
-  client.query(
-    query,
-    [ database ],
-    function (err, result) {
-      if (err) {
-        log.trace( { op: 'MySql.createSchema:MakingTheSchema', err: err.message } )
-        return d.reject(err)
-      }
-      d.resolve(result[0].count === 0 ? false : true)
-    }
-  )
-  return d.promise
-}
-
-function readDbPatchLevel(dbMetadataExists) {
-  log.trace( { op: 'MySql.createSchema:ReadDbPatchLevel' } )
-  if ( dbMetadataExists === false ) {
-    // the table doesn't exist, so start at patch level 0
-    return P.resolve(0)
-  }
+var patch = require('../db/patch')
 
-  // find out what patch level the database is currently at
-  var d = P.defer()
-  var query = "SELECT value FROM dbMetadata WHERE name = ?"
-  client.query(
-    query,
-    [ options.patchKey ],
-    function(err, result) {
-      if (err) {
-        log.trace( { op: 'MySql.createSchema:ReadDbPatchLevel', err: err.message } )
-        return d.reject(err)
-      }
-      // convert the patch level from a string to a number
-      return d.resolve(+result[0].value)
-    }
-  )
-  return d.promise
-}
+console.log(patch)
 
-function patchToRequiredLevel(currentPatchLevel) {
-  log.trace( { op: 'MySql.createSchema:PatchToRequiredLevel' } )
+// set some options
+var options = options.master
+options.dir = path.join(__dirname, '..', 'db', 'schema')
+options.patchKey = 'schema-patch-level'
```

Comment:
There's a config option for this, we should read it from there rather than hard-coding.